csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -780,6 +783,62 @@ private static void HexEncodeUtf16CodeUnit(TextWriter writer, char c)
             writer.Write(Hex[(c >> 4) & 0xf]);
             writer.Write(Hex[(c >> 0) & 0xf]);
         }
+        
+        
+        private void WriteJsonObjectOpen(TextWriter writer, char openChar = '{')
+        {
+            writer.Write(openChar);
+            if (!settings.Pretty)
+            {
+                writer.Write(' ');
+            }
+        }
+
+        private void WriteJsonObjectClose(TextWriter writer, bool hasFields, int depth, char closeChar = '}')
+        {
+            if (hasFields)
+            {
+                if (settings.Pretty)
+                {
+                    writer.WriteLine();
+                    WriteIndentation(writer, depth);
+                }
+                else
+                {
+                    writer.Write(" ");
+                }
+            }
+            
+            writer.Write(closeChar);
+        }
+
+        private void WriteJsonObjectEntry(TextWriter writer, bool first, int depth)
```

Comment:
I'm definitely fine with the implementation, but let's noodle on the name. How about "WriteEntryPrefix"? It's not always an actual separator, and it's not writing the entry itself. (I considered "value" instead of "entry", but when writing object properties the value is only the second part.)